### ISSUE TITLE
Test failures for BITOP lshift on TCL 8.5

### ISSUE THREAD

**Issue by bpo** [JohnSully/KeyDB#99]
I'm seeing a single test consistently failing in Amazon Linux. I'm guessing this is a problem with mismatched requirements on my build machine but was surprised to see the same failure when I tried a clean checkout on my MacOS laptop.

Amazon Linux 2 output:
```
[err]: BITOP lshift string in tests/unit/bitops.tcl
Expected '�g' to equal or match 'abcdefg'
```

MacOS output:
```
*** [err]: BITOP lshift string in tests/unit/bitops.tcl
Expected 'ïg' to equal or match 'abcdefg'
```

Both setups built and tested correctly for v5.0.2. I'll try to look at this more closely once I have some spare time.

**Comment by JohnSully** [comment]
Hi bpo,

I don’t repro these on my machine but I will definitely try. Can you tell me which AWS instance your seeing this on?  Is it ARM?

**Comment by bpo** [comment]
m5.xlarge, happens on v5.1.0, v5.1.1 and unstable, possibly libc vs
jemalloc?

**Comment by bpo** [comment]
Ok I can reproduce this consistently. It's much more mundane than a malloc issue.

TCL 8.6:
```
% puts "\x00abcdefg"
abcdefg
```

TCL 8.5:
```
% puts "\x00abcdefg"
ïg
```

**Comment by JohnSully** [comment]
Ah. Perhaps I didn’t write the test in portable TCL, its not a language I know as well as I would like.  It looks like this is a test only issue though which is much better.

**Comment by JohnSully** [comment]
The magic line in https://www.tcl.tk/man/tcl8.4/TclCmd/encoding.htm: 

> However, because the source command always reads files using the current system encoding, Tcl will only source such files correctly when the encoding used to write the file is the same.

So the encoding of string literals is dependent on the default system encoding.  I need to explicitly indicate it if I want to do fancy things like encode NUL chars.  Should be quite easy to fix.

**Comment by bpo** [comment]
8.5 and 8.6 return different results even when `encoding system` is `utf-8` in both for me. Seems their version of NUL is special: https://wiki.tcl-lang.org/page/utf-8

**Comment by bpo** [comment]
so maybe just specify iso8859

**Comment by JohnSully** [comment]
Fixed in change 5e40ea6ee7f0f34a9e11fdd7518d81383dd73c41

Confirmed working with TCL 8.5 and 8.6 (on Amazon Linux 2, and Ubuntu 18.04 respectively).
